scan-command/src/test/java/io/ballerina/scan/internal/Rule001.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan.internal;
+
+import io.ballerina.projects.Document;
+import io.ballerina.scan.Issue;
+import io.ballerina.scan.RuleKind;
+import io.ballerina.scan.utils.RuleDescription;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+/**
+ * Checkpanic expression analyzer tests.
+ *
+ * @since 0.1.0
+ */
+public class Rule001 extends StaticCodeAnalyzerTest {
+
+    @Test(description = "test checkpanic analyzer")
+    void testCheckpanicAnalyzer() {
+        String documentName = "rule001_rule_checkpanic.bal";
+        Document document = loadDocument(documentName);
+        ScannerContextImpl scannerContext = new ScannerContextImpl(List.of(CoreRule.AVOID_CHECKPANIC.rule()));
+        StaticCodeAnalyzer staticCodeAnalyzer = new StaticCodeAnalyzer(document,
+                scannerContext, document.module().getCompilation().getSemanticModel());
+        staticCodeAnalyzer.analyze();
+        List<Issue> issues = scannerContext.getReporter().getIssues();
+        Assert.assertEquals(issues.size(), 1);
+
+        assertIssue(issues.get(0), documentName, 20, 17, 20, 39, "ballerina:1", 1,
```

Comment:
Shouldn't the line and column numbers here be 21, 18, 21, 28? Do we need to check other places as well?

Reply:
> Shouldn't the line and column numbers here be 21, 18, 21, 28? Do we need to check other places as well?

In here the columns and lines start from 0

Reply:
But we should assert the position the user reads in the code. Similar to how we assert compile-time errors. I believe we fixed this for the reports anyway.

Reply:
yes.
Shall I create a issue since it is a separate task?

Reply:
Created https://github.com/ballerina-platform/static-code-analysis-tool/issues/29

Reply:
Can't we fix this instead? Can happen in the assertion util function. Or we'll have to go back and change all tests later.

Reply:
ok I will do that.

Reply:
I will update this with the very next PR